### PR TITLE
Remove hanging ties on XML import

### PR DESCRIPTION
**Pull request by miiizen**
This PR ports some of https://github.com/musescore/MuseScore/pull/8556, but doesn't replace hanging ties with l.v. articulations, opting instead to simply delete them pending a proper feature. 

The PR also limits the length of ties. Ties ending in a different bar will only be connected if there are no intermediary notes or rests in a voice which avoids this case (the final A has a tie ending tag). 
image

Dolet 6 doesn't export cross stave information which can lead to unterminated ties. We search for these cases, and if found create cross stave ties:
screenshot_2024-04-12_at_09 38 22_720

### CODE REVIEW

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -7450,30 +7520,12 @@ static void addArpeggio(ChordRest* cr, String& arpeggioType, int arpeggioNo, Arp
     }
 }
 
-//---------------------------------------------------------
-//   addArticLaissezVibrer
-//---------------------------------------------------------
-
-static void addArticLaissezVibrer(const Note* const note)
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7457]: removing it here...

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -7513,14 +7568,37 @@ static void addTie(const Notation& notation, const Score* score, Note* note, con
...
+                LOGI() << "Discard";
+                cleanupUnterminatedTie(ties[note->pitch()], score, fixForCrossStaff);
+            }
+            ties[note->pitch()] = nullptr;
+        } else {
+            logger->logError(String("Non-started tie terminated. No-op."), xmlreader);
+        }
         // ignore
     } else if (type == "let-ring") {
         addArticLaissezVibrer(note);
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7604]: ...doesn't go well here

**Review by oktophonie — APPROVED** [review]

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 3]
```diff
@@ -1598,6 +1598,62 @@ SpannerSet MusicXMLParserPass2::findIncompleteSpannersAtPartEnd()
...
+//   addArticLaissezVibrer
+//---------------------------------------------------------
+
+static void addArticLaissezVibrer(const Note* const note)
+{
+    IF_ASSERT_FAILED(note) {
+        return;
+    }
+
+    auto chord = note->chord();
```
Comment (mike-spa) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:1611]: I'm not a fan of "auto". I use it only for complicated types like iterators etc, but when the type is as simple as Chord*, better make it explicit

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 4]
```diff
@@ -1598,6 +1598,62 @@ SpannerSet MusicXMLParserPass2::findIncompleteSpannersAtPartEnd()
...
+    if (fixForCrossStaff) {
+        const Segment* nextSeg = score->tick2leftSegment(unterminatedChord->tick() + unterminatedChord->ticks());
+        if (nextSeg) {
+            const Part* part = unterminatedTieNote->part();
+            for (track_idx_t track = part->startTrack(); track <= part->endTrack(); ++track) {
+                const EngravingItem* el = nextSeg->element(track);
+                if (el && el->isChord()) {
+                    Note* matchingNote = toChord(el)->findNote(unterminatedTieNote->pitch());
+                    if (matchingNote && matchingNote->tpc1() == unterminatedTieNote->tpc1()
+                        && matchingNote->tpc2() == unterminatedTieNote->tpc2()) {
```
Comment (mike-spa) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:1642]: I don't think you need to check both tpc1 and tpc2. As far as I know, the first is the concert pitch version, the latter is the transposed version: if one is equal, the other must be equal too. Perhaps you can simply use note->tpc(), which will choose by itself

**Review by mike-spa — APPROVED** [review]

**Pull request merged**